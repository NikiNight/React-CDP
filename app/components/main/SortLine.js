import React from 'react';

const SortLine = (props) => {

    return (
        <div className="sort-line">
            <div className="wrapper">
                <p className="sort-line__movies-num">
                    {props.moviesNum} movies found
                </p>
                {
                    props.moviesNum>1 &&
                    <div className="sort-line__selectors">
                        Sort by 
                        <span 
                            className={`button_minimal ${props.activeSort === 'Date' ? 'button_minimal_active' : ''}`} 
                            onClick={() => props.toogleSorting('Date')}
                        >release date</span>
                        <span 
                            className={`button_minimal ${props.activeSort === 'Rating' ? 'button_minimal_active' : ''}`}
                            onClick={() => props.toogleSorting('Rating')}
                        >raiting</span>
                    </div>
                }
            </div>
        </div>
    )

}

export default SortLine;