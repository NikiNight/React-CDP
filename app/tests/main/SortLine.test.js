import React from 'react';
import SortLine from './../../components/main/SortLine';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

describe('Test SortLine component', () => {

    test('SortLine matches snapshot', () => {
        const tree = renderer.create(<SortLine />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should be call toogleSorting', () => {
        const toogleSorting = jest.fn();
        const component = mount(<SortLine toogleSorting={toogleSorting} moviesNum={2} activeSort="Date"/>);
        component.find('.button_minimal_active').simulate('click');
        expect(toogleSorting).toHaveBeenCalled();
    });

    it('Should be call toogleSorting with parametr', () => {
        const toogleSorting = jest.fn();
        const component = mount(<SortLine toogleSorting={toogleSorting} moviesNum={2} activeSort="Date"/>);
        component.find('.button_minimal:not(.button_minimal_active)').simulate('click');
        expect(toogleSorting).toHaveBeenCalledWith('Rating');
    });

    it('Inactive button should have appropriate content', () => {
        const toogleSorting = jest.fn();
        const component = mount(<SortLine toogleSorting={toogleSorting} moviesNum={2} activeSort="Rating"/>);
        const button = component.find('.button_minimal:not(.button_minimal_active)');
        expect(button.text()).toBe('release date');
    });

})