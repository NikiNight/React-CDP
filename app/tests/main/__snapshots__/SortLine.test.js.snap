// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SortLine component SortLine matches snapshot 1`] = `
<div
  className="sort-line"
>
  <div
    className="wrapper"
  >
    <p
      className="sort-line__movies-num"
    >
       movies found
    </p>
  </div>
</div>
`;
